### PR TITLE
Adds error handling if authentication fails

### PR DESCRIPTION
**Pull request by hannahkwarren**
Commit message(s) added to this PR:
Adds error handling if authentication fails

Why is this change necessary (explain like I'm 5)?
If the user credentials provided aren't valid, we need to respond with an error.

What changed technically that may impact others? Step-by-step instructions to test that it's working as intended.
we added some logic to SignInMutation to respond with an error on failed login.

Why did we make this change? What Changed? How do we test it?
SignInMutation and the associated test changed. Test with provided spec!

### CODE REVIEW

File: app/graphql/mutations/sign_in_musician.rb [hunk 1]
```diff
@@ -14,16 +14,16 @@ def resolve(credentials: nil)
...
-
-      len = ActiveSupport::MessageEncryptor.key_len
-      salt  = SecureRandom.random_bytes(len)
-      key   = ActiveSupport::KeyGenerator.new('password').generate_key(salt, len)
-      crypt = ActiveSupport::MessageEncryptor.new(key)
-      token = crypt.encrypt_and_sign("musician-id:#{ musician.id }")
-
-      { musician: musician, token: token }
+      if musician && musician.authenticate(credentials[:password])
+        len = ActiveSupport::MessageEncryptor.key_len
```
Comment (omegaeye-he) [inline app/graphql/mutations/sign_in_musician.rb:18]: all these variables looks really clutter and hard to read in the `if`. What do you think about moving them above the `if` or into an `initialize`?

File: spec/graphql/mutations/sign_in_musician_spec.rb [hunk 2]
```diff
@@ -23,4 +23,28 @@ def query
...
-end
\ No newline at end of file
+
+  it 'errors on failure to authenticate' do
+    musician_2 = Musician.create!(email: 'bruce@mail.com', password: 'password', name: 'Phil Brazil', photo: 'www.photo.com', phone: '5597995111')
+    post '/graphql', params: { query: query2 }
+    response_body = JSON.parse(response.body, symbolize_names: true)
+    
+    expect(response_body[:data][:signInMusician]).to eq(nil)
+    expect(response_body[:errors][0][:message]).to eq('Invalid login credentials')
```
Comment (omegaeye-he) [inline spec/graphql/mutations/sign_in_musician_spec.rb:33]: I would expect a status code too.

File: app/graphql/mutations/sign_in_musician.rb (hunk 1 above)
Comment (hannahkwarren) [reply app/graphql/mutations/sign_in_musician.rb]: Good point! Moved them outside the if block.

File: spec/graphql/mutations/sign_in_musician_spec.rb (hunk 2 above)
Comment (hannahkwarren) [reply spec/graphql/mutations/sign_in_musician_spec.rb:33]: GraphQL only ever responds with a 200 Status, but we checked the docs and there's an "extension" key we added with a :code that is now "UNAUTHORIZED".

**Pull request merged**